app/views/users/_form.html.erb

```diff
@@ -0,0 +1,57 @@
+<div class="container mx-auto min-h-screen w-1/2 bg-gray-100 py-12 px-4 sm:px-6 lg:px-8">
+  <% flash.each do |key, value| %>
+  <%= content_tag :div, value, class: "bg-gray-700 text-white p-4 #{key}" %>
+<% end %>
+  <%= form_for @user, html: { class: "w-full" } do |f| %>
+  <ul>
+    <% @user.errors.full_messages.each do |error| %>
+      <li><%= error %></li>
+    <% end %>
+  </ul>
+  <%= f.label :username, class: "block text-gray-700 mt-4 mb-1" %>
+  <%= f.text_field :username, placeholder: "johnDoe", class: "sappearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:shadow-outline-blue focus:border-blue-300 focus:z-10 sm:text-sm sm:leading-5" %>
+  <%= f.label(:email, "Email Address: ", class: "block text-gray-700 mt-4 mb-1" ) %>
+  <%= f.email_field :email, placeholder: 'myemail@provider.tld', class: "sappearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:shadow-outline-blue focus:border-blue-300 focus:z-10 sm:text-sm sm:leading-5" %>
+  <%= f.label :password, "Password: ", class: "block text-gray-700 mt-4 mb-1" %>
+  <%= f.password_field :password, placeholder: 'Your Password', class: "sappearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:shadow-outline-blue focus:border-blue-300 focus:z-10 sm:text-sm sm:leading-5" %>
+  <%= f.label :password_confirmation, "Confirm Password", class: "block text-gray-700 mt-4 mb-1" %>
+  <%= f.password_field :password_confirmation, placeholder: 'Please Confirm Your Password', class: "sappearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:shadow-outline-blue focus:border-blue-300 focus:z-10 sm:text-sm sm:leading-5" %>
+  <%= f.submit "Sign Up", class: "mt-4 cursor-pointer relative flex justify-center py-2 px-4 border border-transparent text-sm leading-5 font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-500 focus:outline-none focus:border-indigo-700 focus:shadow-outline-indigo active:bg-indigo-700 transition duration-150 ease-in-out" %>
+<% end %>
+</div> 
+
+
+
+<%
+=begin%>
+ <%= form_tag users_path do %>
+<%= label_tag :email, 'Email: ' %>
+<%= email_field_tag :email  %>
+<%= label_tag :username, "Username: " %>
+<%= text_field_tag :username  %>
+<%= label_tag :password, "Password: " %>
+<%= password_field_tag :password  %>
+<%= label_tag :password_confirmation, "Confirm Password: " %>
+<%= password_field_tag :password_confirmation  %>
+<%= submit_tag "Sign Up" %>
+<% end %> 
+<%
+=end%>
+
+
+ <%
+=begin%>
+ <form action="<%= users_path %>" method="post" accept-charset="UTF-8">
```

Comment:
This could be refactored as follows:
```ruby
<form action=users_path method='post' accept-charset="UTF-8">
```

Reply:
@stratospherique thanks for your comment. when we do like the below
```suggestion
 <form action=users_path method="post" accept-charset="UTF-8">
```
the system reads a `users_path` variable as a string.

<img width="876" alt="Screen Shot 2020-08-26 at 12 11 20 PM" src="https://user-images.githubusercontent.com/38358081/91286299-f03f1c80-e796-11ea-97d2-c30295737671.png">